### PR TITLE
feat: adapt Xero tap to work with private applications (DM-351)

### PR DESCRIPTION
**Pull request by mdimercurio**
Exclude streams that will not be used

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -9,11 +9,11 @@
       classifiers=["Programming Language :: Python :: 3 :: Only"],
       py_modules=["tap_xero"],
       install_requires=[
-          "python-dateutil==2.6.0",  # This is required by singer-python,
+          "python-dateutil",  # This is required by singer-python,
```
Comment (jacobfrackson) [inline setup.py]: these should, in theory, still be pinned

Comment (jacobfrackson) [reply setup.py]: maybe not for this one but definitely for singer-python and pyxero I think

**Review by jacobfrackson — CHANGES_REQUESTED** [review]
This works but it would be better to have the config somewhere else. Also, because we don't have one repo for this I think we should have documentation around this on Notion for style guide and stuff.

File: README.md [hunk 2]
```diff
@@ -36,9 +36,21 @@ This tap:
...
 
-## Limitations
+## Running the tap
 
- - Only designed to work with Xero [Partner Applications](https://developer.xero.com/documentation/auth-and-limits/partner-applications), not Private Applications.
+- Create config file with the following variables:
+    - consumer_key
+    - client_secret
+    - start_date
+- Validate that the private key is located at the following path: `~/.ssh/privatekey.pem`
```
Comment (jacobfrackson) [inline README.md]: What private key? If I were running this would I need to generate one of these?

File: tap_xero/credentials.py [hunk 3]
```diff
@@ -48,13 +52,16 @@ def download_from_s3(config):
...
+    with open(str(Path.home()) + '/.ssh/privatekey.pem') as keyfile:
+        rsa_key = keyfile.read()
+
     return OAuth1(
-        config["consumer_key"],
-        client_secret=config["consumer_secret"],
-        resource_owner_key=config["oauth_token"],
-        resource_owner_secret=config["oauth_token_secret"],
-        rsa_key=config["rsa_key"],
+        config['consumer_key'],
```
Comment (jacobfrackson) [inline tap_xero/credentials.py]: single quotes, double quotes, single quotes 🙄

File: tap_xero/streams.py [hunk 4]
```diff
@@ -178,39 +178,39 @@ def sync(self, ctx):
...
 
     # NON-PAGINATED STREAMS
     # These endpoints do not support pagination, but do support the Modified At
     # header.
-    BookmarkedStream("accounts", ["AccountID"]),
-    BookmarkedStream("bank_transfers", ["BankTransferID"], bookmark_key="CreatedDateUTC"),
-    BookmarkedStream("employees", ["EmployeeID"]),
-    BookmarkedStream("expense_claims", ["ExpenseClaimID"]),
-    BookmarkedStream("items", ["ItemID"]),
+    Everything("accounts", ["AccountID"]),
```
Comment (jacobfrackson) [inline tap_xero/streams.py]: Does this work?

**Review by jacobfrackson — CHANGES_REQUESTED** [review]

File: tap_xero/streams.py (hunk 4 above)
Comment (mdimercurio) [reply tap_xero/streams.py]: yes

File: README.md (hunk 2 above)
Comment (mdimercurio) [reply README.md]: this is the application's key generated from Xero. I will add a comment to clarify this

File: tap_xero/streams.py (hunk 4 above)
Comment (mdimercurio) [reply tap_xero/streams.py]: reversed to `BookmarkerStream` since the issue was with `start_date` used in the config file

**Pull request merged**